### ISSUE TITLE
inference line 61 only use  one token to predict a new one? not a sentence?

### ISSUE THREAD

**Issue by zhangbo2008** [deepseek-ai/DeepSeek-V3#595]

**Comment by zhangbo2008** [comment]
logits = model.forward(tokens[:, :cur_pos], prev_pos)  is it corret?

**Comment by GeeeekExplorer** [comment]
It is corret. Your can for [this blog](https://medium.com/@joaolages/kv-caching-explained-276520203249) for more details about KV Cache.

**Comment by zhangbo2008** [comment]
> It is corret. Your can for [this blog](https://medium.com/@joaolages/kv-caching-explained-276520203249) for more details about KV Cache.

thanks a lot. but i still  think it is wrong
